2.7.2.Final does not start when smallrye-metrics is on classpath

### Describe the bug

Hello

I'm attempting to get client MP metrics per https://quarkus.io/guides/micrometer#quarkus-micrometer_quarkus.micrometer.binder.mp-metrics.enabled and https://quarkus.io/guides/smallrye-metrics

I am not using resteasy reactive but plain resteasy

When I add implementation("io.quarkus:quarkus-smallrye-metrics") as a dependency Quarkus does not start and throws below

Bug looks similar as https://github.com/apache/camel-quarkus/issues/2442 .

Is it a known bug ? I Did not find related issues 

```
2022-03-01 11:17:39,857 ERROR [io.qua.dep.dev.IsolatedDevModeMain] (main) Failed to start quarkus: java.lang.RuntimeException: io.quarkus.builder.ChainBuildException: Multiple producers of item class io.quarkus.deployment.metrics.MetricsCapabilityBuildItem (io.quarkus.smallrye.metrics.deployment.SmallRyeMetricsProcessor#metricsCapabilityBuildItem)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:330)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:252)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:60)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.firstStart(IsolatedDevModeMain.java:92)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:455)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:66)
	at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:140)
	at io.quarkus.bootstrap.app.CuratedApplication.runInAugmentClassLoader(CuratedApplication.java:96)
	at io.quarkus.deployment.dev.DevModeMain.start(DevModeMain.java:132)
	at io.quarkus.deployment.dev.DevModeMain.main(DevModeMain.java:62)
Caused by: io.quarkus.builder.ChainBuildException: Multiple producers of item class io.quarkus.deployment.metrics.MetricsCapabilityBuildItem (io.quarkus.smallrye.metrics.deployment.SmallRyeMetricsProcessor#metricsCapabilityBuildItem)
	at io.quarkus.builder.BuildStepBuilder.build(BuildStepBuilder.java:198)
	at io.quarkus.builder.BuildStepBuilder.buildIf(BuildStepBuilder.java:212)
	at io.quarkus.deployment.ExtensionLoader.lambda$loadStepsFromClass$102(ExtensionLoader.java:817)
	at java.base/java.util.function.Consumer.lambda$andThen$0(Consumer.java:65)
	at io.quarkus.deployment.ExtensionLoader.lambda$loadStepsFromClass$103(ExtensionLoader.java:910)
	at java.base/java.util.function.Consumer.lambda$andThen$0(Consumer.java:65)
		at java.base/java.util.function.Consumer.lambda$andThen$0(Consumer.java:65)
	at java.base/java.util.function.Consumer.lambda$andThen$0(Consumer.java:65)
	at java.base/java.util.function.Consumer.lambda$andThen$0(Consumer.java:65)
	at java.base/java.util.function.Consumer.lambda$andThen$0(Consumer.java:65)
(snip, repeated dozens of times)
java.base/java.util.function.Consumer.lambda$andThen$0(Consumer.java:65)
	at java.base/java.util.function.Consumer.lambda$andThen$0(Consumer.java:65)
	at java.base/java.util.function.Consumer.lambda$andThen$0(Consumer.java:65)
	at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:107)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:328)
	... 9 more
Caused by: java.lang.Throwable: This is the location of the conflicting producer (io.quarkus.micrometer.deployment.MicrometerProcessor#metricsCapabilityBuildItem).
	at io.quarkus.builder.BuildStepBuilder.build(BuildStepBuilder.java:198)
	at io.quarkus.builder.BuildStepBuilder.buildIf(BuildStepBuilder.java:212)
	at io.quarkus.deployment.ExtensionLoader.lambda$loadStepsFromClass$102(ExtensionLoader.java:817)
	at java.base/java.util.function.Consumer.lambda$andThen$0(Consumer.java:65)
	at io.quarkus.deployment.ExtensionLoader.lambda$loadStepsFromClass$103(ExtensionLoader.java:910)
	... 138 more




```

### Expected behavior

Quarks boots when I enable smallrye metrics

### Actual behavior

Does not boot

### How to Reproduce?

Add io.quarkus:quarkus-smallrye-metrics as a dependency

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

java 11

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.7.2.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

Does not work on 2.6.3 either. Untested below.

--- a/extensions/micrometer/runtime/pom.xml
+++ b/extensions/micrometer/runtime/pom.xml
@@ -128,6 +128,11 @@
             <plugin>
                 <groupId>io.quarkus</groupId>
                 <artifactId>quarkus-bootstrap-maven-plugin</artifactId>
+                <configuration>
+                    <capabilities>
+                        <provides>io.quarkus.metrics</provides>
+                    </capabilities>
+                </configuration>
             </plugin>
             <plugin>
                 <artifactId>maven-compiler-plugin</artifactId>

